src/Shipping_Method/PostNL.php

```diff
@@ -49,19 +49,51 @@ public function postnl_init() {
 		$this->init_form_fields();
 		$this->init_settings();
 
-		add_filter( 'woocommerce_shipping_instance_form_fields_' . $this->id, array( $this, 'change_title_default' ), 10, 1 );
+		add_filter( 'woocommerce_shipping_instance_form_fields_' . $this->id, array( $this, 'instance_form_fields' ), 10, 1 );
 		add_action( 'woocommerce_update_options_shipping_' . $this->id, array( $this, 'process_admin_options' ) );
 	}
 
+	/**
+	 * Calculate the shipping costs.
+	 *
+	 * @param array $package Package of items from cart.
+	 */
+	public function calculate_shipping( $package = array() ) {
+		// Set free shipping rate if cart subtotal exceed minimum_for_free_shipping
+		$minimum_for_free_shipping = $this->get_option( 'minimum_for_free_shipping' );
+		if ( '' !== $minimum_for_free_shipping && $package['cart_subtotal'] > $minimum_for_free_shipping ) {
+			$rate = array(
+				'id'      => $this->get_rate_id(),
+				'label'   => $this->title,
+				'cost'    => 0,
+				'package' => $package,
+			);
+
+			$this->add_rate( $rate );
+		} else {
+			parent::calculate_shipping();
+		}
+	}
+
 	/**
 	 * Manipulate the title default.
```

Comment:
```suggestion
	 * Add form fields for PostNL.
```
Let's update the comment as well.